ui/src/shared/utils/useDragBehavior.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react'
+import {useCallback} from 'react'
+
+export interface DragEvent {
```

Comment:
suprised this type doesn't already exist?

Reply:
Hmm we have something like it in giraffe (though incompatible). Maybe that's what you're thinking of?